src/platform/plugins/shared/discover/public/application/main/discover_main_route.tsx

```diff
@@ -147,6 +150,123 @@ const DiscoverMainRouteContent = (props: SingleTabViewProps) => {
+  }> => {
+    const attachments: Array<{
+      id: string;
+      type: string;
+      getContent: () => Promise<Record<string, unknown>> | Record<string, unknown>;
+    }> = [];
+
+    // Screen context attachment
+    attachments.push({
+      id: 'discover-screen-context',
+      type: AttachmentType.screenContext,
+      getContent: async () => {
+        const currentTabRuntimeState = selectTabRuntimeState(runtimeStateManager, currentTabId);
+        const stateContainer = currentTabRuntimeState?.stateContainer$.getValue();
+
+        if (!stateContainer) {
+          return {
+            app: 'discover',
+            description: 'Discover - Data exploration',
+            additional_data: {},
+          };
+        }
+
+        const appState = stateContainer.appState.getState();
+        const savedSearch = stateContainer.savedSearchState.getState();
+        const currentTab = stateContainer.getCurrentTab();
+        const timeRange = currentTab.globalState.timeRange;
+        const isEsqlMode = isDataSourceType(appState.dataSource, DataSourceType.Esql);
+        const isSaved = Boolean(savedSearch.id);
+
+        const additionalData: Record<string, string> = {
+          isSaved: isSaved ? 'true' : 'false',
+          mode: isEsqlMode ? 'esql' : 'normal',
+        };
+
+        if (isSaved && savedSearch.id) {
+          additionalData.savedSearchId = savedSearch.id;
+        }
+
+        if (timeRange) {
+          additionalData.timeRange = JSON.stringify({
+            from: timeRange.from,
+            to: timeRange.to,
+          });
+        }
+
+        if (isEsqlMode && isOfAggregateQueryType(appState.query)) {
+          additionalData.esqlQuery = appState.query.esql;
+        }
+
+        if (appState.columns && appState.columns.length > 0) {
+          additionalData.selectedFields = JSON.stringify(appState.columns);
+        }
+
+        return {
+          app: 'discover',
+          description: 'Discover - Data exploration',
+          url: window.location.href,
+          additional_data: additionalData,
+        };
+      },
+    });
+
+    // Text attachment with first 25 documents
+    attachments.push({
+      id: 'discover-documents',
+      type: AttachmentType.text,
+      getContent: async () => {
+        const currentTabRuntimeState = selectTabRuntimeState(runtimeStateManager, currentTabId);
+        const stateContainer = currentTabRuntimeState?.stateContainer$.getValue();
+
+        if (!stateContainer) {
+          return { content: '' };
+        }
+
+        // Get current documents from the data state
+        const documents$ = stateContainer.dataState.data$.documents$;
+        const documents = documents$.getValue();
+
+        if (!documents || !documents.result || documents.result.length === 0) {
+          return { content: 'No documents available' };
+        }
+
+        // Get first 25 documents
+        const first25Docs = documents.result.slice(0, 5);
```

Comment:
why does `0,5` get 25 docs? Are there 5 docs in each slice?

Reply:
nah, i limited it down to 5 later as 25 was too much :D ignore all changes in discover/ app, they will be removed when i make this ready for review, its just to showcase how this gets used and to make it easier to test.